app/views/ideas/show.html.erb

```diff
@@ -0,0 +1,28 @@
+<!-- show one idea and all its comments, can upvote main idea and comments, can click to reply to idea or comment, can click on square to see author of idea or comment -->
+
+<% if current_user %>
+  <%= button_to("↑ #{@idea.votes.count}", votes_path, params: { votable_id: @idea.id, votable_type: "Idea", idea: @idea.id}, :class => "idea-vote-button") %> <!-- ajax will grab this and create a vote -->
+<% end %>
+
+
+<h1><%= @idea.title %></h1>
+<p id="idea-description"><%= @idea.description %> <%= link_to("■", user_path(@idea.creator)) %></p>
+
+<% if current_user %>
+  <p><%= link_to("Yes! And…", new_comment_path(commentable_type: "Idea", commentable_id: @idea.id), remote: true, id: "idea_reply", :"data-current-id" => @idea.id) %></p>
+<% end %>
+
+<div class="comments">
+  <% @idea.comments.reverse.each do |comment| %>
+
+  <% if current_user %>
+    <%= button_to("↑ #{comment.votes.count}", votes_path, params: { votable_id: comment.id, votable_type: "Comment", idea: @idea.id }, :class => "comment-vote-button") %>
+    <p><%= comment.text %> <%= link_to("Yes! And…", new_comment_path(commentable_type: "Comment", commentable_id: comment.id), remote: true, class: "comment_reply") %></p>
```

Comment:
Ruby doesn't have any strict practices on how long your lines can be, and the community argues a lot about what is best. In general, I encourage people to keep them at or below 100ish characters. To read this line of code in github I have to scroll horizontally and can't see it all on the screen at once. Nobody is taking away points if your files are long, as long as your code is simple and easy to read.